docs/dev/rust.md

```diff
@@ -7,32 +7,43 @@ shows how to use them in Scylla.
 
 To create a Rust package `new_pkg` and use it in a C++ source file:
 1. Create a new package in the `rust` directory using `cargo new new_pkg --lib`
-2. Modify crate type to a static library by adding
+2. Add `new_pkg = { path = "new_pkg", version = "0.1.0" }` to the dependencies list in `rust/Cargo.toml`
+3. Add `extern crate new_pkg;` to `rust/src/lib.rs`
+4. Configure your package in `new_pkg/Cargo.toml` and write your Rust code in `new_pkg/src/lib.rs` (and other `new_pkg/src/*.rs` files)
+5. To export a function `fn foo(x: i32) -> i32` in the namespace `xyz`, add its declaration to `new_pkg/src/lib.rs` as follows:
 ```
-[lib]
-crate-type = ["staticlib"]
-```
-to `new_pkg/Cargo.toml`
-3. Add `"new_pkg",` to the workspace members list in `Cargo.toml`
-4. Write your Rust code in `new_pkg/src/lib.rs`
-5. To export a function `fn foo(x: i32) -> i32`, add its declaration as follows to the same file
-```
-#[cxx::bridge(namespace = "rust")]
+#[cxx::bridge(namespace = "xyz")]
 mod ffi {
     extern "Rust" {
         fn inc(x: i32) -> i32;
     }
 }
 ```
-6. Add `"rust/new_pkg/src/lib.rs"` to the `rusts` list in `configure.py`
-7. Include the `rust/new_pkg.hh` header and use `rust::foo()` in the selected c++ file.
+6. Add `new_pkg/src/lib.rs` to the `configure.py` dependencies where you'll need the Rust exports
+7. Include the `rust/new_pkg.hh` header and use `xyz::foo()` in the selected c++ file.
```

Comment:
I added an option to put the cxx::bridge into other files to the `configure.py`.
I mentioned this option as an annotation and left the base guide unchanged because you need to satisfy some extra conditions to be able to use it and it would make the guide too complicated if we added exact descriptions of all of them to it.